Chore: escape quotes in release action

katspaugh opened the pull request:
## What it solves

The release action pastes the contents of the RC pull request into a bash variable using an echo command, but if the PR description contains a quote it breaks the bash syntax.

E.g.
```
Remove "What's new"
```

usame-algan commented on .github/workflows/tag-release.yml:
```diff
@@ -3,7 +3,7 @@ name: Tag & GitHub release
   pull_request_target:
     branches:
-      - main
+      - dev
```
Why the change from main to dev?

katspaugh commented:
After trying a few different things, the best solution was to pass the body as a variable instead of a file. For that, I've replaced our release action which also fixed the obsolete set_output warnings.
